lib/pages/movies_page.dart

```diff
@@ -0,0 +1,164 @@
+          Padding(
+            padding: const EdgeInsets.all(rowPadding),
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                MoviePoster(
+                  pathToPosterImg: movie.pathToPosterImg,
+                  width: posterWidth,
+                  height: posterHeight,
+                ),
+                Container(width: imageSpacing),
+                Expanded(
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      Container(height: textSpacing),
+                      Text(
+                        movie.title,
+                        style: const TextStyle(
+                          fontSize: titleFontSize,
+                          color: Colors.white,
+                          fontWeight: FontWeight.bold,
+                        ),
+                      ),
+                      Container(height: textSpacing),
+                      Text(
+                        releaseYear,
+                        style: TextStyle(
+                          color: Colors.grey[300],
+                        ),
+                      ),
+                      Container(height: textSpacing),
+                      Wrap(
+                        children: [
+                          for (String genre
+                              in movie.genres.take(displayedGenresAmount))
+                            buildGenreCard(genre),
+                        ],
+                      ),
+                      Center(
+                        child: Transform.scale(
+                          scale: scoreIconScale,
+                          child: MovieScore(
+                            movieScore: movie.score,
+                            voteCount: movie.voteCount,
+                          ),
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Card buildGenreCard(String genre) {
+    return Card(
+      color: UIConstants.genreCardColor,
+      child: Container(
+        decoration: BoxDecoration(
+          borderRadius: const BorderRadius.all(
+            Radius.circular(UIConstants.genreCardBorderRadius),
+          ),
+          border: Border.all(
+            color: UIConstants.genreCardBorderColor,
+            width: UIConstants.genreCardBorderWidth,
+          ),
+        ),
+        child: Padding(
+          padding: const EdgeInsets.all(genreCardPadding),
+          child: Text(
+            genre,
+            style: const TextStyle(
+              color: UIConstants.genreCardTextColor,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+        future: MovieRepository().getMovies(),
```

Comment:
We can add some Flutter best practices here. Please read this:
https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html

> If the future is created at the same time as the FutureBuilder, then every time the FutureBuilder's parent is rebuilt, the asynchronous task will be restarted. A general guideline is to assume that every build method could get called every frame, and to treat omitted calls as an optimization.

TLDR, you could make this widget `Stateful`, declare a `Future<T> movies` at the beginning, init that future in the `initState` method, and use that future in the `FutureBuilder`. Like here:
https://stackoverflow.com/a/68966640

Reply:
nice, thats good to know. Fixed!